### ISSUE TITLE
Arch Linux Compile Failure in idlib

### ISSUE THREAD

**Issue by thebenenator** [RobertBeckebans/RBDOOM-3-BFG#371]
Attempt to compile latest git (did not try older versions) quickly fails.

```
[  0%] Creating idlib/precompiled.h.gch for idlib
In file included from ../idlib/Lib.h:250:0,
                 from precompiled.h:48:
../idlib/Heap.h:72:1: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
 throw( std::bad_alloc ) // DG: standard signature seems to include throw(..)
 ^~~~~
../idlib/Heap.h:87:1: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
 throw( std::bad_alloc ) // DG: standard signature seems to include throw(..)
 ^~~~~
In file included from ../idlib/Lib.h:295:0,
                 from precompiled.h:48:
../idlib/StrStatic.h: In constructor ‘idStrStatic<_size_>::idStrStatic(bool)’:
../idlib/StrStatic.h:100:23: error: no matching function for call to ‘idStrStatic<_size_>::operator=(const bool&)’
   idStr::operator=( b );
                       ^
In file included from ../idlib/Lib.h:294:0,
                 from precompiled.h:48:
../idlib/Str.h:674:16: note: candidate: void idStr::operator=(const idStr&)
 ID_INLINE void idStr::operator=( const idStr& text )
                ^~~~~
../idlib/Str.h:674:16: note:   no known conversion for argument 1 from ‘const bool’ to ‘const idStr&’
../idlib/Str.h:153:10: note: candidate: void idStr::operator=(const char*)
  void    operator=( const char* text );
          ^~~~~~~~
../idlib/Str.h:153:10: note:   no known conversion for argument 1 from ‘const bool’ to ‘const char*’
In file included from ../idlib/Lib.h:295:0,
                 from precompiled.h:48:
../idlib/StrStatic.h: In constructor ‘idStrStatic<_size_>::idStrStatic(char)’:
../idlib/StrStatic.h:108:23: error: no matching function for call to ‘idStrStatic<_size_>::operator=(const char&)’
   idStr::operator=( c );
                       ^
In file included from ../idlib/Lib.h:294:0,
                 from precompiled.h:48:
../idlib/Str.h:674:16: note: candidate: void idStr::operator=(const idStr&)
 ID_INLINE void idStr::operator=( const idStr& text )
                ^~~~~
../idlib/Str.h:674:16: note:   no known conversion for argument 1 from ‘const char’ to ‘const idStr&’
../idlib/Str.h:153:10: note: candidate: void idStr::operator=(const char*) <near match>
  void    operator=( const char* text );
          ^~~~~~~~
../idlib/Str.h:153:10: note:   conversion of argument 1 would be ill-formed:
In file included from ../idlib/Lib.h:295:0,
                 from precompiled.h:48:
../idlib/StrStatic.h: In constructor ‘idStrStatic<_size_>::idStrStatic(int)’:
../idlib/StrStatic.h:116:23: error: no matching function for call to ‘idStrStatic<_size_>::operator=(const int&)’
   idStr::operator=( i );
                       ^
In file included from ../idlib/Lib.h:294:0,
                 from precompiled.h:48:
../idlib/Str.h:674:16: note: candidate: void idStr::operator=(const idStr&)
 ID_INLINE void idStr::operator=( const idStr& text )
                ^~~~~
../idlib/Str.h:674:16: note:   no known conversion for argument 1 from ‘const int’ to ‘const idStr&’
../idlib/Str.h:153:10: note: candidate: void idStr::operator=(const char*) <near match>
  void    operator=( const char* text );
          ^~~~~~~~
../idlib/Str.h:153:10: note:   conversion of argument 1 would be ill-formed:
In file included from ../idlib/Lib.h:295:0,
                 from precompiled.h:48:
../idlib/StrStatic.h: In constructor ‘idStrStatic<_size_>::idStrStatic(unsigned int)’:
../idlib/StrStatic.h:124:23: error: no matching function for call to ‘idStrStatic<_size_>::operator=(const unsigned int&)’
   idStr::operator=( u );
                       ^
In file included from ../idlib/Lib.h:294:0,
                 from precompiled.h:48:
../idlib/Str.h:674:16: note: candidate: void idStr::operator=(const idStr&)
 ID_INLINE void idStr::operator=( const idStr& text )
                ^~~~~
../idlib/Str.h:674:16: note:   no known conversion for argument 1 from ‘const unsigned int’ to ‘const idStr&’
../idlib/Str.h:153:10: note: candidate: void idStr::operator=(const char*) <near match>
  void    operator=( const char* text );
          ^~~~~~~~
../idlib/Str.h:153:10: note:   conversion of argument 1 would be ill-formed:
In file included from ../idlib/Lib.h:295:0,
                 from precompiled.h:48:
../idlib/StrStatic.h: In constructor ‘idStrStatic<_size_>::idStrStatic(float)’:
../idlib/StrStatic.h:132:23: error: no matching function for call to ‘idStrStatic<_size_>::operator=(const float&)’
   idStr::operator=( f );
                       ^
In file included from ../idlib/Lib.h:294:0,
                 from precompiled.h:48:
../idlib/Str.h:674:16: note: candidate: void idStr::operator=(const idStr&)
 ID_INLINE void idStr::operator=( const idStr& text )
                ^~~~~
../idlib/Str.h:674:16: note:   no known conversion for argument 1 from ‘const float’ to ‘const idStr&’
../idlib/Str.h:153:10: note: candidate: void idStr::operator=(const char*)
  void    operator=( const char* text );
          ^~~~~~~~
../idlib/Str.h:153:10: note:   no known conversion for argument 1 from ‘const float’ to ‘const char*’
In file included from ../framework/../sys/sys_localuser.h:31:0,
                 from ../framework/../sys/sys_session.h:32,
                 from ../framework/UsercmdGen.h:32,
                 from precompiled.h:66:
../framework/../sys/sys_profile.h: At global scope:
../framework/../sys/sys_profile.h:65:7: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
  std::auto_ptr< idSaveGameProcessorSaveProfile > profileSaveProcessor;
       ^~~~~~~~
In file included from /usr/include/c++/7.1.1/memory:80:0,
                 from sys/sys_includes.h:148,
                 from precompiled.h:34:
/usr/include/c++/7.1.1/bits/unique_ptr.h:51:28: note: declared here
   template<typename> class auto_ptr;
                            ^~~~~~~~
In file included from ../framework/../sys/sys_localuser.h:31:0,
                 from ../framework/../sys/sys_session.h:32,
                 from ../framework/UsercmdGen.h:32,
                 from precompiled.h:66:
../framework/../sys/sys_profile.h:66:7: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
  std::auto_ptr< idSaveGameProcessorLoadProfile > profileLoadProcessor;
       ^~~~~~~~
In file included from /usr/include/c++/7.1.1/memory:80:0,
                 from sys/sys_includes.h:148,
                 from precompiled.h:34:
/usr/include/c++/7.1.1/bits/unique_ptr.h:51:28: note: declared here
   template<typename> class auto_ptr;
                            ^~~~~~~~
make[2]: *** [idlib/CMakeFiles/precomp_header_idlib.dir/build.make:58: idlib/CMakeFiles/precomp_header_idlib] Error 1
make[1]: *** [CMakeFiles/Makefile2:200: idlib/CMakeFiles/precomp_header_idlib.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

GCC 7.1.1, binutils 2.28.0, mesa 17.1.2. It's 64-bit, though going by the "no matching function for call" errors, I doubt that has anything to do with it.

**Comment by damiel** [comment]
Hey, i encountered the same problem on myy arch installation and managed to come up with a possible fix, i created a pull request but for a fork of rbdoom3bfg called opentechbfg. If you get a chance can you maybe apply the same changes to rbdoom3bfg and test if the changes fix the issue for you aswell?

You can check out my pull request here: https://github.com/OpenTechEngine/OpenTechBFG/pull/136

**Comment by j0hnnybash** [comment]
@damiel, tried your fix for this version here, and yeah, works ;)

**Comment by hollunder** [comment]
Yep, please fix :)

**Comment by shmerl** [comment]
Not limited to Arch. Same happens to me on Debian testing.

**Comment by shmerl** [comment]
By the way. What's the difference between this repo, and one at OpenTechEngine? Which one should I use?

**Comment by BielBdeLuna** [comment]
OTE is a fork from RBDoom3BFG made with the intention of having an OS engine prepared to make games, so there is no doom classic and doom 2 code there and we're working on a minimum of free base assets so people can mod it into a game.

**Comment by hollunder** [comment]
Is there a straightforward way to create a patch from this pull request?

**Comment by damiel** [comment]
Eh, guess i will just go ahead and port my changes also to rbdoom3bfg, shouldnt take to long.

**Comment by shmerl** [comment]
@BielBdeLuna: If both RBDOOM-3-BFG and OTE have the same issue, where do you recommend to open bugs or submit patches?

**Comment by MadDeCoDeR** [comment]
Fixed it in [Classic RBDoom3BFG](https://github.com/MadDeCoDeR/Classic-RBDOOM-3-BFG)

**Comment by pbtoast** [comment]
This should be fixed by #442 in master now.
